src/protocol/ldap/org/apache/jmeter/protocol/ldap/sampler/TrustAllSocket.java

```diff
@@ -0,0 +1,117 @@
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TrustAllSocket extends SocketFactory {
+    private static final AtomicReference<TrustAllSocket> defaultFactory = new AtomicReference<>();
+
+    private static final Logger log = LoggerFactory.getLogger(TrustAllSocket.class);
+
+    private static SSLSocketFactory sf;
+
+    public TrustAllSocket() throws NoSuchAlgorithmException, KeyManagementException {
+        TrustManager[] trustAllCerts = new TrustManager[] {new X509ExtendedTrustManager() {
+            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+            public void checkClientTrusted(X509Certificate[] certs, String authType) {
+            }
+            public void checkServerTrusted(X509Certificate[] certs, String authType) {
+            }
+            @Override
+            public void checkClientTrusted(X509Certificate[] arg0, String arg1, Socket arg2)
+                throws CertificateException {
+            }
+            @Override
+            public void checkClientTrusted(X509Certificate[] arg0, String arg1, SSLEngine arg2)
+                throws CertificateException {
+            }
+            @Override
+            public void checkServerTrusted(X509Certificate[] arg0, String arg1, Socket arg2)
+                throws CertificateException {
+            }
+            @Override
+            public void checkServerTrusted(X509Certificate[] arg0, String arg1, SSLEngine arg2)
+                throws CertificateException {
+            }
+        }};
+
+        SSLContext ctx = null;
+        ctx = SSLContext.getInstance("TLS");
+        ctx.init(null, trustAllCerts, new java.security.SecureRandom());
+        sf = ctx.getSocketFactory();
+    }
+
+    public static SocketFactory getDefault() {
+        final TrustAllSocket value = defaultFactory.get();
+        if (value == null) {
+            try {
+                defaultFactory.compareAndSet(null, new TrustAllSocket());
+            } catch (KeyManagementException e) {
+                log.error("KeyManagementException: "+e.getLocalizedMessage());
+            } catch (NoSuchAlgorithmException e) {
+                log.error("NoSuchAlgorithmException: "+e.getLocalizedMessage());
+            }
+            return defaultFactory.get();
+        }
+        return value;
+    }
+    @Override
+    public Socket createSocket() throws IOException{
+        return sf.createSocket();
+    }
+
+    @Override
+    public Socket createSocket(final String s, final int i) throws IOException {
+        return sf.createSocket(s, i);
+    }
+
+    @Override
+    public Socket createSocket(final String s, final int i, final InetAddress inetAddress, final int i1) throws IOException {
```

Comment:
For readability could we please name the parameters the same as `SocketFactory`
```suggestion
    public Socket createSocket((String host, int port, InetAddress localHost, int localPort) throws IOException
```

Reply:
completed